File: src/main/java/com/developersam/web/model/ten/TenClientMove.java
package com.developersam.web.model.ten;

/**
 * A data class that represents a client move.
 */
public class TenClientMove {
    
    /**
     * The board data before the human move.
     */
    final TenBoardData boardBeforeHumanMove;
    /**
     * The new human move.
     */
    final int[] humanMove;
    
    /**
     * The default no arg constructor for GSON.
     */
    private TenClientMove() {
        boardBeforeHumanMove = null;
        humanMove = null;
    }
    
}

File: developersam-ten/src/main/kotlin/com/developersam/ten/TenClientMove.kt
package com.developersam.ten

/**
 * A data class that represents a client move.
 * - [boardBeforeHumanMove] specifies the board before the human move.
 * - [humanMove] completes the picture by providing human's move in a tuple.
 */
class TenClientMove(
        internal val boardBeforeHumanMove: TenBoardData,
        internal val humanMove: IntArray
)